sklearn/utils/_array_api.py

```diff
@@ -165,8 +165,11 @@ def _expit(X):
     return 1.0 / (1.0 + xp.exp(-X))
 
 
-def _asarray_with_order(array, dtype=None, order=None, copy=None, xp=None):
-    """Helper to support the order kwarg only for NumPy-backed arrays
+def _asarray_with_order(
+        array, dtype=None, order=None, copy=None, _asarray_fn=None, xp=None
```

Comment:
Since we don't impose keywords only, it would be safer to place it at the last position even if this is a private function.